<launch>
    <arg name="fcu_url" default="/dev/ttyUSB0:230400"/>
    <arg name="gcs_url" default="udp://@192.168.50.254"/>
    <arg name="SYS_ID" default="1"/>
    <arg name="X" default="0.25"/>
	<arg name="Y" default="0.25"/>    



    <group ns="MAV$(arg SYS_ID)">
        <param name="UAV_ID" value="$(arg SYS_ID)"/>
        <param name="obs_safe_D" value="0.6"/>
        <param name="obs_gamma" value="0.8"/>
        <param name="MAV_safe_D" value="0.6"/>
        <param name="MAV_gamma" value="0.8"/>
        <param name="delay_step" value="0"/>
		<param name="cbf" type="bool" value="false"/>
		<param name="hover_x" value="$(arg X)"/>
		<param name="hover_y" value="$(arg Y)"/>

        <include file="$(find mavros)/launch/apm.launch">
            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg SYS_ID)"/>
        </include> 
		
        <node name="topic_tool" pkg="flight_control" type="topic_tool"></node>
        <node name="origin_py" pkg="flight_control" type="origin.py" output="screen"></node>
        <node name="velocity_cbf_generator" pkg="flight_control" type="velocity_cbf_generator" output="screen">
            <rosparam subst_value="true">
                    use_input: "velocity"  #velocity,position
            </rosparam>
        </node>
    </group>
</launch>
